### ISSUE TITLE
False Positive S4325/no-useless-cast: singleton types 

### ISSUE THREAD

**Issue by andrea-guarino-sonarsource** [SonarSource/SonarTS#650]
```javascript
let a = <"aValue">"aValue"
```
Inferred type would be "string", hence the cast is no useless.

**Comment by artola** [comment]
Similar case:

```ts
const Elevation = {
  ZERO: 0 as 0,
  ONE: 1 as 1,
  TWO: 2 as 2,
  THREE: 3 as 3,
  FOUR: 4 as 4,
};
type Elevation = typeof Elevation[keyof typeof Elevation];
```

**Comment by vilchik-elena** [comment]
Can be done together with #482
